crates/std_detect/src/detect/os/linux/auxvec.rs

```diff
@@ -73,7 +76,10 @@ pub(crate) struct AuxVec {
 pub(crate) fn auxv() -> Result<AuxVec, ()> {
     #[cfg(all(
         feature = "std_detect_dlsym_getauxval",
-        not(all(target_os = "linux", target_env = "gnu")),
+        not(all(
+            target_os = "linux",
+            any(target_env = "gnu", target_env = "musl", target_env = "ohos"),
+        )),
         // TODO: libc crate currently doesn't provide getauxval on 32-bit Android.
         not(all(target_os = "android", target_pointer_width = "64")),
     ))]
```

Comment:
There's a lot of duplication in this function which makes it hard to review. I think this is a good opportunity to move the choice of whether to use `libc::getauxval` or look it up dynamically into the local `getauxval` function.

Reply:
Done in https://github.com/rust-lang/stdarch/pull/1746/commits/2960b074140080208da46870a96ead4b7d03ae35.

I actually found a difference between the two blocks...

https://github.com/rust-lang/stdarch/blob/5ba56c7924058623d230f39f9073c1e42c775775/crates/std_detect/src/detect/os/linux/auxvec.rs#L83-L89

https://github.com/rust-lang/stdarch/blob/5ba56c7924058623d230f39f9073c1e42c775775/crates/std_detect/src/detect/os/linux/auxvec.rs#L130-L137